### DISCUSSION TITLE
re-using reverse geocache data

### DISCUSSION THREAD

**Discussion by computamike** [openbenches/openbenches.org#347] [Ideas]
I guess it might be possible that there are a number of benches in a location.  Rather than hitting up a geocoding API - if we have already resolved that latitude and longitude in the database, then we could just re-use that?  

Potentially you could loosen the confines of the coordinate check - so if there is a bench within a certain threshold of an existing bench (say  - within 10 meters) then you could potentially use that address?

**Comment by edent** [comment]
We could do. But the problem is one of borders. Two benches 10 meters apart could be in different countries.  Borders can also change fairly frequently, so re-using would have to be time-limited.

There are also some commercial restrictions about storing geo-coded responses - but I'm looking to see if I can find a provider which can accommodate us.

**Comment by computamike** [comment]
sorry - I re-read you reply.  I'm not sure I fully understand.  My thoughts on this were : 

If I am at point 1 and pop in a bench, then the system will hit an API based on the lat/long of the photo I just took.  It generates an address from that, and stores that in the database.

Now - if I am still at point 1, and I take a picture of the bench next door to it, then the whole process would be repeated?  But I would suggest that instead you could search the Open Benches database - see if there are any benches that we have stored within 2 meters of the location (hey - we just stored a bench right next door to it), and if so we could copy the address from that bench for this new bench.  So - in this instance we would save a hit the API.

**Comment by edent** [comment]
I'm not sure what that would gain? The API call is quick and inexpensive. We can get 3,000 lookups per day from https://www.geoapify.com/pricing - I don't think we've ever exceeded that. We can also get 170,000 per day from https://geocode.maps.co/

There is a small cost if we use StadiaMaps - but their TOS prevent us from storing the result of the address lookup.

The issue of proximity is a problem. Take this bench, for example, https://openbenches.org/bench/23191 - a couple of meters away is a different country. Labelling a bench from one side of a border as being in the other is inaccurate at best, and offensive at worst.
